docs/reference/design-patterns.md

```diff
@@ -0,0 +1,363 @@
+---
+type: doc
+layout: reference
+category: "Basics"
+title: "Design Patterns"
+---
+
+# Design Patterns
+
+This is a list of Java design patterns that should be implemented differently in Kotlin than in Java.
+Design patterns that are the same for Kotlin and Java are not listed here.
+
+
+## Natively supported patterns
+
+Patterns that have dedicated Kotlin syntax for their support.
+
+### Singleton
+
+``` java
+public class Singleton {
+    String data = "some usefull shared data";
```

Comment:
but in [Kotlin-version](https://github.com/JetBrains/kotlin-web-site/pull/442/files#diff-17261ec8f99c1daa36b1928b4b9b11f3R44) this field is immutable